spec/controllers/challenge_assignments_controller_spec.rb

```diff
@@ -15,12 +15,63 @@
   end
 
   context "when logged in" do
+    let(:gift_exchange) { create(:gift_exchange) }
+    let(:collection2) { create(:collection, challenge: gift_exchange, challenge_type: "GiftExchange") }
```

Comment:
I think you can move some of the lets inside their specific describes, and avoid using number suffixes.